src/directives.ts

```diff
@@ -0,0 +1,136 @@
+
+import { getClosest } from './utils';
+
+
+interface DestroyHTMLElement extends HTMLElement {
+    $destroy: () => void;
+}
+
+interface DestroyHTMLInputElement extends HTMLInputElement {
+  $destroy: () => void;
+}
+
+/**
+ * Triggers action when scroll gets close to the bottom.
+ *
+ * Example usage:
+ * <div v-infinite-scroll="action()">Scroll me</div>
+ */
+const infiniteScroll: DirectiveOptions = {
+  bind(el: DestroyHTMLElement, binding: DirectiveBinding): void {
+    const tolerance = parseInt(binding.arg || '100', 10);
+    const action = binding.value;
+
+    el.addEventListener('scroll', (): void => {
+      const height = (el.firstChild as DestroyHTMLElement).clientHeight;
+      if (el.scrollTop >= height - el.clientHeight - tolerance) {
+        action();
+      }
+    });
+  },
+};
+
+/**
+ * Scroll input into view when focused.
+ *
+ * Example usage:
+ * <input type="text" value="foo" v-scroll-on-focus />
+ */
+const scrollOnFocus: DirectiveOptions = {
+  bind(el: DestroyHTMLInputElement): void {
+    const target = el.querySelector('input,textarea');
+    if (!target) {
+      return;
+    }
+
+    const scrollTo = (target: HTMLElement, scrollable: HTMLElement, scrollableContent: HTMLElement, marginTop: number = 0): void => {
+      while (target.offsetParent && target.offsetParent !== scrollable) {
+        marginTop -= target.offsetTop;
+        target = target.offsetParent as HTMLElement;
+      }
+      scrollableContent.scrollTop = target.offsetTop - marginTop;
+    };
+
+    const handler = (): void => {
+      const scrollable = getClosest(el, '.scrollable');
+      const scrollableContent = getClosest(el, '.scrollable-content');
+      if (scrollable && scrollableContent) {
+        const height = scrollableContent.offsetHeight;
+        setTimeout((): void => {
+          // if scrollableContent height is reduced in half second
+          // since an input got focus we assume soft keyboard is showing.
+          if (height > scrollableContent.offsetHeight) {
+            scrollTo(target as HTMLElement, scrollable, scrollableContent, 20);
+          }
+        }, 500);
+      }
+    };
+
+    target.addEventListener('focus', handler);
+    el.$destroy = (): void => {
+      target.removeEventListener('focus', handler);
+    };
+  },
+  unbind(el: DestroyHTMLElement): void {
+    el.$destroy();
+  },
+};
+
+export interface TrackClickDataAttributes {
+  'event-name'?: string;
+  'event-props'?: {[key: string]: any}, // eslint-disable-line @propelinc/no-explicit-any
+}
+
+export type trackClickHandlerFunction = (dataAttributes: TrackClickDataAttributes) => void;
+
+interface TrackClickDirectiveOptions extends DirectiveOptions {
+  handler: trackClickHandlerFunction;
+}
+
+/**
+ * Track an event to amplitude on click.
+ *
+ * Example usage:
+ * <div v-track-click event-name="foo" :event-props="{ bar: 'tzar' }">Click me</div>
+ */
+const trackClick: TrackClickDirectiveOptions = {
+  handler() {
+    return;
+  },
```

Comment:
this handler bs is to allow the users of this library function a way to overwrite what should happen on trackClick. This should allow us to avoid having amplitude as a dependency for the phoenix library